doc/wol/Wake-on-LAN-HLD.md

```diff
@@ -0,0 +1,136 @@
+| gNOI         | **g**RPC **N**etwork **O**perations **I**nterface |
+| NIC          | **N**etwork **I**nterface **C**ontroller          |
+
+## Overview
+
+**Wake-on-LAN** (**WoL** or **WOL**) is an Ethernet or Token Ring computer networking standard that allows a computer to be turned on or awakened from sleep mode by a network message[^1]. This document describes the Wake-on-LAN feature design in SONiC.
+
+## Background
+
+Below diagram describes a common usage of WoL on SONiC switch, this HLD will focus on the green part:
+
+![WoL Background](./wol-background.png)
+
+## Magic Packet
+
+**Magic Packet** is a Ethernet frame with structure:
+
+* **Ethernet Frame Header**:
+  * **Destination MAC address**: broadcast MAC address (ff:ff:ff:ff:ff:ff) or target device's MAC address. [6 bytes]
+  * **Source MAC address**. [6 bytes]
+  * **EtherType**: `0x0842`. [2 bytes]
+* **Ethernet Frame Payload**:
+  * Six bytes of all `0xff`. [6 bytes]
+  * Sixteen repetitions of the target device's MAC address. [96 bytes]
+  * (Optional) A four or six byte password. [4 or 6 bytes]
+
+```
+Byte     |
+Offset   |0     |1     |2     |3     |4     |5     |
+---------+------+------+------+------+------+------+
+       0 |             Destination MAC             |
+         +------+------+------+------+------+------+        ETHERNET FRAME
+       6 |               Source MAC                |           HEADER
+         +------+------+---------------------------+
+      12 | 0x08 | 0x42 |   \      \      \      \
+         +------+------+------+------+------+------+------------------------------
+      14 | 0xFF | 0xFF | 0xFF | 0xFF | 0xFF | 0xFF |
+         +------+------+------+------+------+------+        ETHERNET FRAME
+      20 | Target MAC (repeat 16 times, 96 bytes)  |           PAYLOAD
+         +-----------------------------------------+
+     116 |    Password (optional, 4 or 6 bytes)    |
+         +-----------------------------------------+
+```
+
+## CLI Design
+
+The `wol` command is used to send magic packet to target device.
+
+This command requires root privilege because it'll modify the power-mode of target device.
+
+### Usage
+
+```
+wol <interface> <target_mac> [-b] [-p password] [-c count] [-i interval]
+```
+
+- `interface`: SONiC interface name.
+- `target_mac`: MAC address of the target device.
+- `-b`: Use broadcast MAC address instead of target device's MAC address as **Destination MAC Address in Ethernet Frame Header**.
+- `-p password`: An optional 4 or 6 byte password, in ethernet hex format or quad-dotted decimal[^3].
+- `-c count`: Count of magic packet to send. `count` must between 1 and 5. Default value is 1. This param must use with `-i`.
+- `-i interval`: Wait `interval` milliseconds between sending each magic packet. `interval` must between 0 and 2000. Default value is 0. This param must use with `-c`.
+
+### Example
+
+```
+admin@sonic:~$ wol Ethernet10 00:11:22:33:44:55
```

Comment:
These commands need to be executed in root mode.
Please update the example accordingly.

Reply:
Thanks for your comment! After consideration, we removed the root privilege requirement.